spec.html

```diff
@@ -28621,6 +28621,191 @@ <h1>
+        </emu-clause>
+
+        <emu-clause id="sec-setsyntheticmoduleexport" type="abstract operation">
+          <h1>
+            SetSyntheticModuleExport (
+              _module_: a Synthetic Module Record,
+              _exportName_: a String,
+              _exportValue_: an ECMAScript language value,
+            ): ~unused~
+          </h1>
+          <dl class="header">
+            <dt>description</dt>
+            <dd>It can be used to set or change the exported value for an existing export of a Synthetic Module Record.</dd>
+          </dl>
+
+          <emu-alg>
+            1. Assert: _module_.[[ExportNames]] contains _exportName_.
+            1. Let _envRec_ be _module_.[[Environment]].
+            1. Assert: _envRec_ is not ~empty~.
+            1. Perform _envRec_.SetMutableBinding(_exportName_, _exportValue_, *true*).
+            1. Return ~unused~.
+          </emu-alg>
+        </emu-clause>
+
+        <emu-clause id="sec-smr-module-record-methods">
+          <h1>Implementation of Module Record Abstract Methods</h1>
+
+          <p>The following are the concrete methods for Synthetic Module Record that implement the corresponding Module Record abstract methods defined in <emu-xref href="#table-abstract-methods-of-module-records"></emu-xref>.</p>
+
+          <emu-clause id="sec-smr-LoadRequestedModules" type="concrete method">
+            <h1>LoadRequestedModules ( ): a Promise</h1>
+            <dl class="header">
+              <dt>for</dt>
+              <dd>a Synthetic Module Record _module_</dd>
+            </dl>
+
+            <emu-alg>
+              1. Return ! PromiseResolve(%Promise%, *undefined*).
+            </emu-alg>
+
+            <emu-note>
+              Synthetic Module Records have no dependencies.
+            </emu-note>
+          </emu-clause>
+
+          <emu-clause id="sec-smr-getexportednames" type="concrete method">
+            <h1>GetExportedNames ( ): a List of Strings</h1>
+            <dl class="header">
+              <dt>for</dt>
+              <dd>a Synthetic Module Record _module_</dd>
+            </dl>
+
+            <emu-alg>
+              1. Return _module_.[[ExportNames]].
+            </emu-alg>
+          </emu-clause>
+
+          <emu-clause id="sec-smr-resolveexport" type="concrete method">
+            <h1>
+              ResolveExport (
+                _exportName_: a String,
+              ): a ResolvedBinding Record or *null*
+            </h1>
+            <dl class="header">
+              <dt>for</dt>
+              <dd>a Synthetic Module Record _module_</dd>
+            </dl>
+
+            <emu-alg>
+              1. If _module_.[[ExportNames]] does not contain _exportName_, return *null*.
+              1. Return ResolvedBinding Record { [[Module]]: _module_, [[BindingName]]: _exportName_ }.
+            </emu-alg>
+          </emu-clause>
+
+          <emu-clause id="sec-smr-Link" type="concrete method">
+            <h1>Link ( ): ~unused~</h1>
```

Comment:
The expectation at the call-sites for `.Link` is that it returns a completion record, so:
```suggestion
            <h1>Link ( ): a normal completion containing ~unused~</h1>
```
and then `Return NormalCompletion(~unused~)` in the algorithm.